cli/src/main/java/com/devonfw/tools/ide/repo/CustomToolMetadata.java

```diff
@@ -0,0 +1,119 @@
+package com.devonfw.tools.ide.repo;
+
+import java.util.Set;
+
+import com.devonfw.tools.ide.os.OperatingSystem;
+import com.devonfw.tools.ide.os.SystemArchitecture;
+import com.devonfw.tools.ide.url.model.file.UrlDownloadFileMetadata;
+import com.devonfw.tools.ide.version.VersionIdentifier;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+/**
+ * Representation of a {@link CustomToolMetadata} from a {@link CustomToolRepository}.
+ */
+public final class CustomToolMetadata implements UrlDownloadFileMetadata {
```

Comment:
Couldn't this be changed to a record too?

Reply:
Surely it could. But what is the value. We have to implement the interface `UrlDownloadFileMetadata` that defines getter methods.
So try it if you like but there is no reasonable benefit in doing so.